### PR TITLE
Most mobs can be mind-swapped

### PR DESCRIPTION
**Pull request by Cocacola4124**
## About The Pull Request
Most mobs can be mind-swapped. Synthetics are blocked, and there is a blacklist for any mobs that cause issues(It's empty). Since eris code is bad a lot of mobs don't work well when directly controlled, but they do work, the player just might shoot gas at random or walk towards a dead body.

## Why It's Good For The Game
You can be a cow.

## Changelog
:cl:
tweak: Most mobs can be mind-swapped into, with a few exceptions.
/:cl:

### CODE REVIEW

**Review by Evankhell561 — CHANGES_REQUESTED** [review]
I recommend you not to make an untested pr.

**Comment by Cocacola4124** [comment]
It was tested, what's the issue?

**Comment by Evankhell561** [comment]
> It was tested, what's the issue?

Did you test this PR?

**Comment by Cocacola4124** [comment]
Reason why it didn't compile initially is because I messed up my github 

> 
> 
> > It was tested, what's the issue?
> 
> Did you test this PR?

Hello?

**Comment by Evankhell561** [comment]
> Reason why it didn't compile initially is because I messed up my github
> 
> > > It was tested, what's the issue?
> > 
> > 
> > Did you test this PR?
> 
> Hello?

answer my question. if you don't understand my bad english i can rephrase it.

**Comment by Cocacola4124** [comment]
Blacklist didn't work,

**Comment by Evankhell561** [comment]
> Blacklist didn't work,

Did you test this pr before publishing it?

**Comment by Cocacola4124** [comment]
> 
> 
> > Blacklist didn't work,
> 
> Did you test this pr before publishing it?

are you blind

**Comment by Evankhell561** [comment]
> > > Blacklist didn't work,
> > 
> > 
> > Did you test this pr before publishing it?
> 
> are you blind

are two different questions.

**Comment by FortuneMaster** [comment]
> Blacklist didn't work,

if it didn't work, why did you open a PR in the first place?

File: code/game/machinery/mindswapper.dm [hunk 1]
```diff
@@ -78,11 +79,10 @@
 	// Get all candidates in range for the mind swapping
 	var/list/swapBoddies = list()
 	var/list/swapMinds = list()
-	for(var/mob/living/carbon/C in range(swap_range,src))
-		if (C.stat != DEAD)  // candidates should not be dead
-			swapBoddies += C
-			swapMinds += C.ghostize(0)
-	
+	for(var/mob/living/M in range(swap_range,src))
+		if (M.stat != DEAD && (M.mob_classification != CLASSIFICATION_SYNTHETIC) && !(M.type in swap_blacklist))  // candidates should not be dead
```
Comment (Evankhell561) [inline code/game/machinery/mindswapper.dm:83]:
```suggestion
		if (M.stat != DEAD && M.mob_classification != CLASSIFICATION_SYNTHETIC && !(M.type in swap_blacklist))  // candidates should not be dead
```

**Comment by Cocacola4124** [comment]
Worked fine, just didn't understand exactly what || did so the blacklist(Which contained only a single mob that never exists) could be swapped into.

**Review by Evankhell561 — CHANGES_REQUESTED** [review]

File: code/game/machinery/mindswapper.dm (hunk 1 above)
Comment (Cocacola4124) [reply code/game/machinery/mindswapper.dm:83]: Doesn't travis complain if you do that?

**Comment by Evankhell561** [comment]
> Worked fine, just didn't understand exactly what || did so the blacklist(Which contained only a single mob that never exists) could be swapped into.

it's ok not to know, it's ok. just try to test your code before publishing the pr. And if you have doubts about those things you can always ask in discord.

File: code/game/machinery/mindswapper.dm [hunk 2]
```diff
@@ -13,6 +13,7 @@
 	var/operating = FALSE  // Is it on?
 	var/swap_time = 200  // Time from starting until minds are swapped
 	var/swap_range = 1
+	var/swap_blacklist = list(/mob/living/simple_animal/hostile/megafauna)
```
Comment (Evankhell561) [inline code/game/machinery/mindswapper.dm:16]:
```suggestion
	var/list/swap_blacklist = list(/mob/living/simple_animal/hostile/megafauna)
```

**Review by Evankhell561 — CHANGES_REQUESTED** [review]

File: code/game/machinery/mindswapper.dm (hunk 2 above)
Comment (Cocacola4124) [reply code/game/machinery/mindswapper.dm]: thank you

File: code/game/machinery/mindswapper.dm [hunk 3]
```diff
@@ -13,6 +13,7 @@
 	var/operating = FALSE  // Is it on?
 	var/swap_time = 200  // Time from starting until minds are swapped
 	var/swap_range = 1
+	var/list/swap_blacklist = list(/mob/living/simple_animal/hostile/megafauna)
```
Comment (TheShown911) [inline code/game/machinery/mindswapper.dm:16]: No fun.

Comment (Cocacola4124) [reply code/game/machinery/mindswapper.dm:16]: This mob never exists, the blacklist exists to block players from swapping into mobs that are just outright broken. It doesn't use istype so only this specific one is blocked, the others like the hivemind megafauna are blocked for control spiders as well.

Comment (Cocacola4124) [reply code/game/machinery/mindswapper.dm:16]: It should work that way at least.

Comment (TheShown911) [reply code/game/machinery/mindswapper.dm:16]: Control spiders check a specific variable to see if they are swapping into a robotics mob, so no robotic mobs work, but if it's not an istype check, it's fine

Comment (Cocacola4124) [reply code/game/machinery/mindswapper.dm:16]: This uses the same check, MOB_CLASSIFICATION to make sure the mob isn't synthetic.

**Review by Firefox13 — APPROVED** [review]

**Pull request merged**